web/modules/weather_data/src/Service/WeatherAlertParser.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+namespace Drupal\weather_data\Service;
+
+class WeatherAlertParser
+{
+    /**
+     * The input description string
+     *
+     * @var descriptionString
+     */
+    private $descriptionString;
+
+    public static function fixupNewlines($str)
+    {
+        if ($str) {
+            // Remove individual newline characters. Leave pairs. Pairs of
+            // newlines are equivalent to paragraph breaks and we want to keep
+            // those, but within a paragraph, we want to let the text break on
+            // its own.
+            return preg_replace("/([^\n])\n([^\n])/m", "$1 $2", $str);
+        }
+        return $str;
+    }
+
+    public function __construct(string $descriptionString)
+    {
+        $this->descriptionString = $descriptionString;
+    }
+
+    public function parse()
```

Comment:
Just curious why use an instance method here and a static one above. Could this be

```php
public static function parse($str)
```

And then call it with `WeatherAlertParser::parse($description);`?

I'm not suggesting let's do it, just curious.

Reply:
@greg-does-weather I went ahead and pushed some changes with what I (probably?) originally intended: https://github.com/weather-gov/weather.gov/pull/748/commits/7512fff23320df33bd55f78342498c1866dd2124
  
LMK what you think

Reply:
👍 LGTM!